client/dashboard/components/switcher/index.tsx

```diff
@@ -0,0 +1,48 @@
+import { Dropdown, Button } from '@wordpress/components';
+import { chevronDownSmall } from '@wordpress/icons';
+import SwitcherContent, { type RenderItemIcon } from './switcher-content';
+import type { PropsWithChildren } from 'react';
+
+export default function Switcher< T >( {
```

Comment:
I extracted the switcher to its own generic component now that we're using it in two places. Let me know what you think of it.

Reply:
I think we should extract it 😃 
There are some a11y issues in the switcher I would love to fix, and I'd much rather fix them in one place.